opm/grid/cpgrid/processEclipseFormat.cpp

```diff
@@ -258,17 +260,44 @@ namespace cpgrid
         std::cout << "Processing eclipse data." << std::endl;
 #endif
         processed_grid output;
-        const size_t global_nc = input_data.dims[0] * input_data.dims[1] * input_data.dims[2];
-        std::vector<int> is_aquifer_cell(global_nc, 0);
-        for ([[maybe_unused]]const auto& [global_index, volume] : aquifer_cell_volumes) {
-            is_aquifer_cell[global_index] = 1;
+        if (ecl_state) {
+            const auto& aquifer = ecl_state->aquifer();
+            const auto aquifer_cell_volumes = aquifer.numericalAquifers().aquiferCellVolumes();
+            const size_t global_nc = input_data.dims[0] * input_data.dims[1] * input_data.dims[2];
+            std::vector<int> is_aquifer_cell(global_nc, 0);
+            for ([[maybe_unused]]const auto&[global_index, volume] : aquifer_cell_volumes) {
+                is_aquifer_cell[global_index] = 1;
+            }
+            process_grdecl(&input_data, z_tolerance, is_aquifer_cell.data(), &output);
+        } else {
+            process_grdecl(&input_data, z_tolerance, nullptr, &output);
         }
-        process_grdecl(&input_data, z_tolerance, is_aquifer_cell.data(), &output);
         if (remove_ij_boundary) {
             removeOuterCellLayer(output);
             // removeUnusedNodes(output);
         }
 
+        if (ecl_state) {
+            assert(deck);
+            const auto& aquifer = ecl_state->aquifer();
+            if (aquifer.hasNumericalAquifer()) {
+                const size_t global_nc = input_data.dims[0] * input_data.dims[1] * input_data.dims[2];
+                std::vector<int> new_actnum(global_nc, 0);
+                for (int i = 0; i < output.number_of_cells; ++i) {
+                    new_actnum[output.local_cell_index[i]] = 1;
```

Comment:
That this line of code actually "turns on" a cell is so special it deserves a comment.

Reply:
Okay. This is not the line turning on a aquifer cell. This line just generates the actnum based on `output`.

Reply:
Here is the place that forcefully keeps aquifer cells active. 

https://github.com/OPM/opm-grid/blob/a28f0bd0dfdf724ee19b3f975efa2c851cd15060/opm/grid/cpgpreprocess/preprocess.c#L354